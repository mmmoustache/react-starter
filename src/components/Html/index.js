import React from 'react';// eslint-disable-line

function MarkUp(markup, data, styles, script, env) {
	return `<!doctype html>
<html lang="en">
<head>
	<meta charset="utf-8">
	<meta http-equiv="x-ua-compatible" content="ie=edge">
	<title>JC Starter</title>
	<meta name="description" content="JC Starter">
	<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
	<link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
	<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
	<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
	<link rel="manifest" href="/site.webmanifest">
	<link rel="mask-icon" href="/safari-pinned-tab.svg" color="#222222">
	<meta name="msapplication-TileColor" content="#ffffff">
	<meta name="theme-color" content="#ffffff">
	${
		env === 'development' ?	`<link rel="stylesheet" href="/client.css" />` : `<style>${styles}</style>`
	}
</head>
<body>
	<div id="root">${markup}</div>
	${
		env === 'development' ?	`<script src="/client.js" async></script>` : `<script>${script}</script>`
	}
	<script>window.__INITIAL_STATE__ = ${JSON.stringify(data)};</script>
</body>
</html>`;
}

export default ({markup, data, styles, script, env}) => {
	const markUp = MarkUp(markup, data, styles, script, env);
	return env === 'development' ?	markUp : markUp.replace(/\n|\t/g, '');
};
