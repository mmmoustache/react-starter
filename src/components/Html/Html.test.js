import React from 'react';
import renderer from 'react-test-renderer';
import Html from '../Html';

describe('Component: ContentWrapper', () => {

  it('Should render successfully', () => {
    const component = renderer.create(
      <Html />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

});