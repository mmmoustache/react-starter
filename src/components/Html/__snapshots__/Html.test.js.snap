// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ContentWrapper Should render successfully 1`] = `"<!doctype html><html lang=\\"en\\"><head><meta charset=\\"utf-8\\"><meta http-equiv=\\"x-ua-compatible\\" content=\\"ie=edge\\"><title>Test</title><meta name=\\"description\\" content=\\"Test\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, shrink-to-fit=no\\"><link rel=\\"apple-touch-icon\\" sizes=\\"180x180\\" href=\\"/apple-touch-icon.png\\"><link rel=\\"icon\\" type=\\"image/png\\" sizes=\\"32x32\\" href=\\"/favicon-32x32.png\\"><link rel=\\"icon\\" type=\\"image/png\\" sizes=\\"16x16\\" href=\\"/favicon-16x16.png\\"><link rel=\\"manifest\\" href=\\"/site.webmanifest\\"><link rel=\\"mask-icon\\" href=\\"/safari-pinned-tab.svg\\" color=\\"#222222\\"><meta name=\\"msapplication-TileColor\\" content=\\"#ffffff\\"><meta name=\\"theme-color\\" content=\\"#ffffff\\"><style>undefined</style></head><body><div id=\\"root\\">undefined</div><script>undefined</script><script>window.__INITIAL_STATE__ = undefined;</script></body></html>"`;
